lib/bsdlib/Kconfig

```diff
@@ -32,6 +32,24 @@ config BSD_LIBRARY_TRACE_ENABLED
 	# This enable UARTE1 peripheral and includes nrfx UARTE driver.
 	select NRFX_UARTE1
 
+config NRF91_SOCKET_SEND_SPLIT_LARGE_BLOCKS
+	bool "Split large blocks passed to send() or sendto()"
+	default n
+	help
+	  Workaround a limitation in the nRF91 modem firmware's bsdlib
+	  implementation regarding return value for send() or sendto()
+	  calls larger than the module can handle. It should send the data
+	  up to the maximum, and return that as the return value. Instead,
+	  it returns error 22.
+
+config NRF91_SOCKET_BLOCK_LIMIT
+	int "Maximum size the modem can send"
+	default 2048
```

Comment:
Where is the default size of 2048 coming from?
The current max size for send/sendto is defined in bsdlib is
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/bsdlib/doc/api.html#c.BSD_IP_MAX_MESSAGE_SIZE
The current value is 4096.

Reply:
4096 does not appear to be the right value during actual testing.  I've written a test program which sends larger and larger blocks; when it crosses the 2048 byte boundary the device crashes.  I can re-verify that and post test results in NRF91-601.